compiler/rustc_mir/src/transform/check_consts/ops.rs

```diff
@@ -268,43 +268,41 @@ impl NonConstOp for CellBorrow {
 }
 
 #[derive(Debug)]
+/// This op is for `&mut` borrows in the trailing expression of a constant
+/// which uses the "enclosing scopes rule" to leak its locals into anonymous
+/// static or const items.
 pub struct MutBorrow(pub hir::BorrowKind);
 
 impl NonConstOp for MutBorrow {
     fn status_in_item(&self, ccx: &ConstCx<'_, '_>) -> Status {
-        // Forbid everywhere except in const fn with a feature gate
-        if ccx.const_kind() == hir::ConstContext::ConstFn {
-            Status::Unstable(sym::const_mut_refs)
-        } else {
-            Status::Forbidden
+        match ccx.const_kind() {
+            // Mutable statics can handle mutable references in their final value
+            hir::ConstContext::Static(hir::Mutability::Mut) => Status::Allowed,
```

Comment:
Shouldn't this *at least* have a feature gate? Also I could not find a testcase for this.
Mutability is tricky, so it's not immediately obvious to me that this is right.^^

Reply:
I undid this, it's not necessary, it was only relevant in the old dataflow scheme, not in this new scheme.